mrp_bom_attribute_match/models/mrp_bom.py

```diff
@@ -14,8 +14,13 @@ class MrpBomLine(models.Model):
     product_backup_id = fields.Many2one(
         "product.product", help="Technical field to store previous value of product_id"
     )
+    bom_product_tmpl_id = fields.Integer(
+        related="bom_id.product_tmpl_id.id", help="Required to avoid BOM recursion"
+    )
```

Comment:
```suggestion
    bom_product_tmpl_id = fields.Many2one(
        related="bom_id.product_tmpl_id", help="Required to avoid BOM recursion"
    )
```

Reply:
Actually, this field is not necessary I think. You can use [`parent_product_tmpl_id`](https://github.com/odoo/odoo/blob/e53f7ef1be9f7a1cc3fd47d30c4a688d95f10760/addons/mrp/models/mrp_bom.py#L351)